tests/test_basic.py

```diff
@@ -437,3 +440,37 @@ def needsUnittestUser(self):
         """Skips the test unless it is run as the ``unittest`` user."""
         if not testonly.is_unittest_user():
             self.skipTest("May fail with non-unittest users.")
+
+
+@pytest.fixture(scope="function")
+def with_metadata_checker(monkeypatch):
```

Comment:
This fixture monkeypatches the urllib3 pool manager to intercept outgoing requests. It's a lightweight alternative to setting up a proxy or some other more complicated software.